src/messages.rs

```diff
@@ -276,12 +325,12 @@ impl Header {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Payload {
     Reduction(payload::Reduction),
     NewBlock(Box<payload::NewBlock>),
     StepVotes(payload::StepVotes),
-    StepVotesWithCandidate(payload::StepVotesWithCandidate),
+    StepVotesWithCandidate(Box<payload::StepVotesWithCandidate>),
```

Comment:
Why did you added the `Box`?

Reply:
Due to a clippy warning for having too big variant.